analyses/methylation-probe-annotations/run-methyl-probe-annotation.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# PediatricOpenTargets 2023
+# Eric Wafula
+set -e
+set -o pipefail
+
+
+# This script should always run as if it were being called from
+# the directory it lives in.
+script_directory="$(perl -e 'use File::Basename;
+ use Cwd "abs_path";
+ print dirname(abs_path(@ARGV[0]));' -- "$0")"
+cd "$script_directory" || exit
+
+
+# Set up paths to input and output directories for the analysis
+absolute_path=$(cd "$(dirname "$0")"; pwd -P)
+data_dir="${absolute_path}/../../data"
+inputs_dir="inputs"
+results_dir="results"
+
+# create tmp/ directory for R scripts
+mkdir -p -m775 ${results_dir}
+
+
+printf '\nStarting Analysis...\n'
+
+
+# Set up paths output files
+gencode_introns_gff="${inputs_dir}/gencode.v39.primary_assembly.introns.annotation.gff3.gz"
+bedtools_intersect="${results_dir}/infinium.gencode.v39.probe.annotations.bedtools.tsv.gz"
+
+
+# add intron coordinates to gencode gff
+gt gff3 -retainids -addintrons \
+  ${inputs_dir}/gencode.v39.primary_assembly.annotation.gff3.gz \
+  | gzip -f \
+  > $gencode_introns_gff 
+
+
+# create gencode features and cpg probe coordinates bed files
+Rscript --vanilla 01-create-bed-files.R \
+  --gencode_gtf ${data_dir}/gencode.v39.primary_assembly.annotation.gtf.gz \
+  --gencode_gff ${inputs_dir}/gencode.v39.primary_assembly.introns.annotation.gff3.gz \
+  --cpg_map ${data_dir}/infinium-methylationepic-v-1-0-b5-manifest-file-csv.zip
```

Comment:
This file is not in the `data/` directory currently

Reply:
> @ewafula this file is not in the `data/` directory currently

@rjcorb , all methylation specifics data release files are not downloaded by default by @jharenza directive - some are too large. 

They are on s3 bucket `v12` and `v13`  folders
md5sum for methyl release files and can download with either curl  or wget 
- curl -O https://d3b-openaccess-us-east-1-prd-pbta.s3.amazonaws.com/open-targets/v13/methyl-md5sum.txt
- curl -O https://d3b-openaccess-us-east-1-prd-pbta.s3.amazonaws.com/open-targets/v13/infinium-methylationepic-v-1-0-b5-manifest-file-csv.zip

I am  having SSL certificate issues on my CHOP laptop, but works ok on EC2
```
curl: (60) SSL certificate problem: self signed certificate in certificate chain
More details here: https://curl.haxx.se/docs/sslcerts.html
```

Reply:
thanks @rjcorb - @ewafula can you put these commands into the bash script or, do a check for those files using the methyl md5sum and if not present in data, then run the download? Thanks!

Reply:
> thanks @rjcorb - @ewafula can you put these commands into the bash script or, do a check for those files using the methyl md5sum and if not present in data, then run the download? Thanks!

Sure, I will.